# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/public/friend/ErrorNumbers.msg
# Commit: 3b62bd57747f
# Full Hash: 3b62bd57747f21c800a93da325d999fa5efb3850
# Author: Jessica Tallon <jtallon@igalia.com>
# Date: 2021-09-03 03:51:58
# Regressor Bug: 1715275
# File Overlap Count: 1
# Description:
#   Bug 1715275 - Add support for a JS WebAssembly.Function object r=rhunt
#   
#   This introduces a new Constructor WebAssembly.Function which is part of
#   the WASM JS-API type reflection work. This allows normal JS functions to
#   be wrapped with a type and used with WebAssembly objects (e.g. Tables).
# ==============================================================================

diff -r f6f4514d09bf -r 3b62bd57747f js/public/friend/ErrorNumbers.msg
--- a/js/public/friend/ErrorNumbers.msg	Thu Sep 02 15:56:01 2021 +0000
+++ b/js/public/friend/ErrorNumbers.msg	Thu Sep 02 15:58:18 2021 +0000
@@ -470,6 +470,8 @@
 MSG_DEF(JSMSG_WASM_BAD_EXN_PAYLOAD,    0, JSEXN_TYPEERR,     "second argument must be an object")
 MSG_DEF(JSMSG_WASM_BAD_EXN_PAYLOAD_LEN, 2, JSEXN_TYPEERR,    "expected {0} values but got {1}")
 MSG_DEF(JSMSG_WASM_BAD_EXN_TAG,        0, JSEXN_TYPEERR,     "exception's tag did not match the provided exception tag")
+MSG_DEF(JSMSG_WASM_BAD_FUNCTION_VALUE, 0, JSEXN_TYPEERR,     "second argument must be a function")
+MSG_DEF(JSMSG_WASM_BAD_ARG_TYPE,       0, JSEXN_TYPEERR,     "parameters and results must be an iterator of value types")
 MSG_DEF(JSMSG_WASM_NO_TRANSFER,        0, JSEXN_TYPEERR,     "cannot transfer WebAssembly/asm.js ArrayBuffer")
 MSG_DEF(JSMSG_WASM_TEXT_FAIL,          1, JSEXN_SYNTAXERR,   "wasm text error: {0}")
 MSG_DEF(JSMSG_WASM_MISSING_MAXIMUM,    0, JSEXN_TYPEERR,     "'shared' is true but maximum is not specified")